asyncgit/src/sync/commit_filter.rs

```diff
@@ -199,14 +203,18 @@ pub fn filter_commit_by_search(
 				.fields
 				.contains(SearchFields::AUTHORS)
 				.then(|| {
-					let name_match = commit
-						.author()
-						.name()
-						.is_some_and(|name| filter.match_text(name));
-					let mail_match = commit
-						.author()
-						.email()
-						.is_some_and(|name| filter.match_text(name));
+					let name_match =
+						get_author_of_commit(&commit, &mailmap)
+							.name()
+							.is_some_and(|name| {
+								filter.match_text(name)
+							});
+					let mail_match =
+						get_author_of_commit(&commit, &mailmap)
+							.email()
+							.is_some_and(|name| {
+								filter.match_text(name)
+							});
 
 					name_match || mail_match
```

Comment:
Maybe we can avoid two calls to `get_author_of_commit`. (This isn't a bottleneck, though.)

```suggestion
					let author =
						get_author_of_commit(&commit, &mailmap);
					[author.email(), author.name()].iter().any(
						|opt_haystack| {
							opt_haystack.is_some_and(|haystack| {
								filter.match_text(haystack)
							})
						},
					)
```